confidence/confidence.go

```diff
@@ -0,0 +1,214 @@
+package confidence
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+	"net/http"
+	"reflect"
+	"strings"
+)
+
+type FlagResolver interface {
+	resolveFlag(ctx context.Context, flag string, defaultValue interface{},
+		evalCtx map[string]interface{}, expectedKind reflect.Kind) InterfaceResolutionDetail
+}
+
+type ContextProvider interface {
+	GetContext() map[string]interface{}
+}
+
+var (
+	SDK_ID      = "SDK_ID_GO_CONFIDENCE"
+	SDK_VERSION = "0.1.8" // x-release-please-version
+)
+
+type Confidence struct {
+	parent        ContextProvider
+	uploader      EventUploader
+	contextMap    map[string]interface{}
+	Config        APIConfig
+	ResolveClient ResolveClient
+}
+
+func (e Confidence) GetContext() map[string]interface{} {
+	currentMap := map[string]interface{}{}
+	parentMap := make(map[string]interface{})
+	if e.parent != nil {
+		parentMap = e.parent.GetContext()
+	}
+for key, value := range parentMap {
```

Comment:
nit: indentation 🫨 

is there something we can add to the CI to check these things?

Reply:
it should be some command as well to do it for the whole project. let me dig a bit.

Reply:
done using `gofmt -w`, also added a step in the ci to check the formatting and fail the step if formatting is not correct.